make/env.mk

```diff
@@ -40,10 +40,14 @@ endif
 TAG := # make sure tag is never injectable as an env var
 RELEASE_GOTAGS := release
 
-# Use a release go -tag when CI is targetting a tag
+# Use a release go -tag when CI is targeting a tag
 ifdef CI
 ifneq ($(BUILD_TAG),)
+ifeq ($(GOTAGS),)
 GOTAGS := $(RELEASE_GOTAGS)
+else
+GOTAGS := $(GOTAGS),$(RELEASE_GOTAGS)
+endif
 endif
 endif
 
```

Comment:
(not about this line)
Independently of which way we go, please add a `# TODO(ROX-1234):` comment next to the new things to eliminate one of env variable or tag (depending on which one is really redundant).